documents/api/viewsets.py

```diff
@@ -30,10 +31,70 @@
     CreateAttachmentSerializer,
     DocumentSerializer,
 )
+from ..serializers.document import DocumentMetadataSerializer
 from ..serializers.status_history import StatusHistorySerializer
-from .docs import attachment_viewset_docs, document_viewset_docs
-from .filtersets import DocumentFilterSet
-from .querysets import get_attachment_queryset, get_document_queryset
+from .docs import (
+    attachment_viewset_docs,
+    document_metadata_viewset_docs,
+    document_viewset_docs,
+)
+from .filtersets import DocumentFilterSet, DocumentMetadataFilterSet
+from .querysets import (
+    get_attachment_queryset,
+    get_document_metadata_queryset,
+    get_document_queryset,
+)
+
+
+@extend_schema_view(**document_metadata_viewset_docs)
+class DocumentMetadataViewSet(AuditLoggingModelViewSet):
```

Comment:
Is this AuditLoggingModelViewSet own code? Could it be GenericViewSet? Not_allowing everything is kind of ugly.

Reply:
AuditLoggingModelViewSet is part of the project. It overrides all those unallowed methods to include logging functionality with API calls. This solution also follows the code style of other previously implemented ViewSets.